auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/LdapFindUserByUidCommand.java

```diff
@@ -1,62 +1,122 @@
 package com.redhat.lightblue.rest.auth.ldap;
 
+import com.google.common.base.Strings;
 import com.netflix.hystrix.HystrixCommand;
 import com.netflix.hystrix.HystrixCommandGroupKey;
 import com.netflix.hystrix.HystrixCommandKey;
 import com.redhat.lightblue.util.ServoGraphiteSetup;
 import org.jboss.logging.Logger;
 
-import javax.naming.NamingEnumeration;
-import javax.naming.directory.SearchControls;
+import javax.naming.NamingException;
 import javax.naming.directory.SearchResult;
 import javax.naming.ldap.LdapContext;
+import java.util.concurrent.ExecutionException;
 
 /**
- * Created by nmalik on 11/11/14.
+ * LDAP Hystrix command that can provide metrics for this service and fall back in case the server was unreachable as well.
+ *
+ * Created by nmalik  and lcestari
  */
 public class LdapFindUserByUidCommand extends HystrixCommand<SearchResult> {
-    private final Logger LOGGER = Logger.getLogger(LightblueLdapRoleProvider.class);
-
     public static final String GROUPKEY = "ldap";
+    private static final String INVALID_PARAM = "%s is null or empty";
+    private static final Logger LOGGER = Logger.getLogger(LightblueLdapRoleProvider.class);
 
     static {
         ServoGraphiteSetup.initialize();
     }
 
-    private final String uid;
-    private final LdapContext ldapContext;
-    private final String ldapSearchBase;
-
+    private final LDAPCacheKey cacheKey;
 
     public LdapFindUserByUidCommand(LdapContext ldapContext, String ldapSearchBase, String uid) {
         super(HystrixCommand.Setter.withGroupKey(HystrixCommandGroupKey.Factory.asKey(GROUPKEY)).
                 andCommandKey(HystrixCommandKey.Factory.asKey(GROUPKEY + ":" + LdapFindUserByUidCommand.class.getSimpleName())));
+        //check if the informed parameters are valid
+        if(Strings.isNullOrEmpty(uid)) {
+            throw new IllegalArgumentException(String.format(INVALID_PARAM, "uid"));
+        } else if(Strings.isNullOrEmpty(ldapSearchBase)) {
+            throw new IllegalArgumentException(String.format(INVALID_PARAM, "ldapSearchBase"));
+        } else if(ldapContext == null) {
+            throw new IllegalArgumentException(String.format(INVALID_PARAM, "ldapContext"));
+        }
 
-        this.ldapContext = ldapContext;
-        this.ldapSearchBase = ldapSearchBase;
-        this.uid = uid;
+        this.cacheKey = new LDAPCacheKey(uid, ldapContext, ldapSearchBase,  "(uid=" + uid + ")");
```

Comment:
https://github.com/lightblue-platform/lightblue-rest/issues/112 created for this